runtime/string_functions.cpp

```diff
@@ -1982,24 +1996,39 @@ static string str_replace_char(char c, const string &replace, const string &subj
 
     piece = pos + 1;
   }
-  php_assert (0);//unreacheable
+  php_assert (0); // unreachable
   return string();
 }
 
-void str_replace_inplace(const string &search, const string &replace, string &subject, int64_t &replace_count) {
-  if ((int)search.size() == 0) {
+static const char *find_substr(const char *where, const char *where_end, const string &what, bool with_case) {
+  if (with_case) {
+    return static_cast<char *>(memmem(where, where_end - where, what.c_str(), what.size()));
+  }
+
+  const auto *it = std::search(where, where_end, what.c_str(), what.c_str() + what.size(), [](char ch1, char ch2) {
```

Comment:
just strcasestr()?

Reply:
Hmm, yes, for sure, it will be better.

Reply:
Hmm, it seems we cannot use `strcasestr`, because we need to search only in `[where, where_end]`.
Perhaps there is a version where we can limit the number of characters, but I did not find one.

Reply:
Done.